ci_framework/roles/polarion/tasks/main.yml

```diff
@@ -0,0 +1,123 @@
+---
+# Copyright Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+- name: Check if variables are defined and not empty
+  ansible.builtin.assert:
+    that:
+      - cifmw_polarion_jump_repo_url is defined and cifmw_polarion_jump_repo_url != ""
+      - cifmw_polarion_test_id is defined and cifmw_polarion_test_id != ""
+      - cifmw_polarion_update_testcases is defined and cifmw_polarion_update_testcases == "false" or cifmw_polarion_update_testcases == "true"
+      - cifmw_polarion_test_results_url is defined and cifmw_polarion_test_results_url != ""
+
+- name: Clone Jump repository
+  ansible.builtin.git:
+    repo: "{{ cifmw_polarion_jump_repo_url }}"
+    dest: "{{ cifmw_polarion_jump_repo_dir }}"
+    version: "{{ cifmw_polarion_jump_repo_branch }}"
+    force: true
+
+- name: Create virtualenv
+  ansible.builtin.pip:
+    name: pip
+    virtualenv: "{{ cifmw_polarion_jump_repo_dir }}/jump-venv"
+    virtualenv_command: "/usr/bin/python3 -m venv"
+    state: latest
+    extra_args: --upgrade
+  when:
+    - cifmw_polarion_jump_repo_dir is defined and cifmw_polarion_jump_repo_dir != ""
+
+# Will be removed before merge
+# - name: Temp task to use polarion-staging
+#   ansible.builtin.replace:
+#     path: "{{ cifmw_polarion_jump_repo_dir }}/{{ item }}"
+#     regexp: "{{ polarion_fqdn }}"
+#     replace: "{{ polarion_stage_fqdn }}"
+#   loop:
+#     - "helpers.py"
+#     - ".pylero"
+
+- name: Run prepare_pylero.sh
+  ansible.builtin.shell: >
+    source jump-venv/bin/activate && \ 
+    ./prepare_pylero.sh
+  args:
+    chdir: "{{ cifmw_polarion_jump_repo_dir }}"
+    executable: /bin/bash
+
+- name: Create {{ cifmw_polarion_jump_result_dir }} directory
+  ansible.builtin.file:
+    path: "{{ cifmw_polarion_jump_result_dir }}"
+    state: directory
+
+- name: Retrieve HTML content
+  ansible.builtin.uri:
+    url: "{{ cifmw_polarion_test_results_url }}"
+    return_content: yes
+  register: html_content
+
+- name: Extract XML URLs
+  ansible.builtin.set_fact:
+    xml_urls: "{{ html_content.content | regex_findall('<a href=\"\\.\\/([^\\/]+\\.xml)\">[^<]+<\\/a>') }}"
```

Comment:
uhhh... can't we avoid that? regex against html are usually a bad idea

Reply:
Changed it to work with fetched artifacts.